### DISCUSSION TITLE
How to build mkdocs with Gitlab ?

### DISCUSSION THREAD

**Discussion by kayielt** [squidfunk/mkdocs-material#6391] [Q&A]
Hello everyone, How can I build mkdocs with Gitlab?  Help me with some advice or do something to do that! thank you!

**Comment by kamilkrzyskow** [comment]
Hello, by "build mkdocs with Gitlab" I understand you want to deploy the generated website to GitLab Pages, right? 

You can find an example of the `.gitlab-ci.yml` file in the documentation: 
https://squidfunk.github.io/mkdocs-material/publishing-your-site/#gitlab-pages
If you want a whole repository example, GitLab provides an official template for MkDocs without the Material theme: 
https://gitlab.com/pages/mkdocs

**Reply by kayielt** [reply]
This is my GitLab with an address like: http://103.129.80.xx/mytest.lnt/edgex-taf_zcu_smartparking but I saw a document in  https://squidfunk.github.io/mkdocs-material/publishing-your-site/#gitlab-pages  , 
`Your documentation should shortly appear at <username>.gitlab.io/<repository>.`
![image](https://github.com/squidfunk/mkdocs-material/assets/151514164/1cf39a68-79eb-409b-a41a-96d637d99ba0). Beside, I got problem when run with CI `"Check runner"`
![image](https://github.com/squidfunk/mkdocs-material/assets/151514164/90096195-deb6-40ef-979b-bd290f7bdb57)

**Reply by kamilkrzyskow** [reply]
> This is my GitLab with an address like: http://103.129.80.xx/mytest.lnt/edgex-taf_zcu_smartparking 

This address looks like an incomplete IP address or a custom `.xx` domain, but I'm guessing it's the former. 
If you can't connect to your GitLab via https://gitlab.com/mytest.lnt/edgex-taf_zcu_smartparking then that http://103.129.80.xx address points to a custom installation of GitLab. 

I have no experience with custom installations, and based on this documentation page https://docs.gitlab.com/ee/administration/pages/source.html I believe you need to add and configure the GitLab Pages onto your installation manually.

> Beside, I got problem when run with CI "Check runner"

The `Job is stuck. Check runners.` warning message indicates that no job runner took the job to process it, therefore it's stuck. 
Continuing with the custom installation idea, you may perhaps have to configure your job runners correctly before they'll take on the jobs. Perhaps they're triggered with specific tags and you need to add those to your CI stages or your installation doesn't have any runners at all.
https://forum.gitlab.com/t/job-is-stuck-check-runners/24381 
https://docs.gitlab.com/ee/ci/yaml/#tags
https://docs.gitlab.com/runner/install/index.html

**Reply by kamilkrzyskow** [reply]
Any updates @kayielt?

**Reply by kayielt** [reply]
Thank you, I will try to do it your way! Maybe in my server, Gitlab does not install "runner" for support build Ci/Cd.
